util/filter.ts

```diff
@@ -0,0 +1,16 @@
+export const Global = -1;
```

Comment:
Small detail, but in most languages the convention for global constants is screaming snake case:

```ts
export const FILTER_GLOBAL = -1;

export const FILTER_DEFAULT_INDEX = -2;
```

(change usages too of course)